### PR TITLE
#141 PR - Login Service Changes/Fix

### LINKED ISSUE #141: Clicking create requires (second) login
After logging in for search, you need to login again if you open create.

### PR DESCRIPTION
**Pull request by hunterwilliams**
#141 targeted
#143 also fixed when adding search to routes.js

Let's discuss what I've got so far... It's going to require possibly some design changes further than what I've done so far

### CODE REVIEW

File: app/templates/ui/app/login/login.service.js [hunk 1]
```diff
@@ -9,9 +9,10 @@
   function LoginService($http, $modal, $q, $rootScope, $state, $stateParams, messageBoardService) {
     var _loginMode = 'full'; // 'modal', 'top-right', or 'full'
     var _loginError;
-    var _isAuthenticated = false;
     var _toStateName;
     var _toStateParams;
+    var _isAuthenticated = isLoggedIn();
```
Comment (grtjn) [inline app/templates/ui/app/login/login.service.js]: Is it your intention to eval the isLoggedIn function right on the spot, or do you want to make isAuthenticated (way down) return a promise from that helper function? (it is doing first, but kinda guessing you want the latter?)

**Comment by grtjn** [comment]
Are any further changes needed in user-service or such? You talked about cyclic dependencies, but this looks clean..

File: app/templates/ui/app/login/login.service.js (hunk 1 above)
Comment (hunterwilliams) [inline app/templates/ui/app/login/login.service.js]: It's currently returning the promise I think. At least the value isn't there in time for the bottom. Ideally I don't want to wait for a promise.

**Comment by hunterwilliams** [comment]
User-service already depends on login-service. That's why I moved over checking status vs making login-service depend on user-service

**Comment by grtjn** [comment]
Yeah, see no reason why login should depend on user service. Looks good to me.. :)

**Comment by hunterwilliams** [comment]
Ill start attempting to make the promise work properly the first go as right now it doesn't resolve in time. I'll also update the tests

File: app/templates/ui/app/login/login.service.spec.js [hunk 2]
```diff
@@ -67,4 +67,60 @@
...
+    }));
+
+    it('should be defined', function () {
+      expect(service).to.be.defined;
+    });
+
+    it('should be logged in already', function() {
+      expect(service.isAuthenticated()).to.be.not.defined;
+      service.getAuthenticatedStatus().then(function(done) {
+        expect(done).to.be.eq(100);
```
Comment (hunterwilliams) [inline app/templates/ui/app/login/login.service.spec.js]: having issues with this this line being ran no matter what...

File: app/templates/ui/app/login/login.service.js [hunk 3]
```diff
@@ -117,20 +154,24 @@
...
-            deregisterLoginSuccess = null;
-            $state.go(next.name, nextParams);
+
+      if (routeIsProtected(next.name)) {
+        var auth = getAuthenticatedStatus();
+
+        if (angular.isFunction(auth.then)) {
+          auth.then(function() {
+            if (!isAuthenticated()) {
+              //this does NOT block requests in a timely fashion...
```
Comment (hunterwilliams) [inline app/templates/ui/app/login/login.service.js]: Got it working so the UI seems correct to the user however it still pushes out requests a bit fast. may need to tweak block route or the login interceptor?

**Comment by grtjn** [comment]
Can @ryanjdew or someone else with more experience with auth in angular review this PR?

**Comment by hunterwilliams** [comment]
Also please let me know what I need to do to fix part of it. See the code comments

**Comment by hunterwilliams** [comment]
My last commit added root.search to protected routes. It never made it back into slush from my sample project apparently. :frowning: Solves issue of "early" requests....

File: app/templates/ui/app/login/login.service.spec.js (hunk 2 above)
Comment (grtjn) [inline app/templates/ui/app/login/login.service.spec.js]: does this still apply with your latest commit?

**Comment by hunterwilliams** [comment]
After some more prodding think it's good. Ready to review & merge :)

**Comment by grtjn** [comment]
Excellent, works good for me. Even when I try to open /profile directly..

**Pull request merged**